Timeout in RunDataFrameAnalyticsIT#testOutlierDetectionStopAndRestart with ml integration tests

A number of failures have arisen lately ([since Apr 4 2020 on master](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-2y,mode:quick,to:now))&_a=(columns:!(branch),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'class:%22org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT%22%20AND%20test:%22testOutlierDetectionStopAndRestart%22%20AND%20branch:master'),sort:!(time,desc)))):

oraclelinux-7: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=oraclelinux-7&&immutable/685/console / https://gradle-enterprise.elastic.co/s/vy7fun2blav4w

centos-8: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos-8&&immutable/685/console / https://gradle-enterprise.elastic.co/s/z2lub544emry6

All of them seem to be failing due to the suite timing out (after `20m`):

```
org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT > testOutlierDetectionStopAndRestart FAILED |  
  | java.lang.Exception: Test abandoned because suite timeout was reached. |  
  | at __randomizedtesting.SeedInfo.seed([591EC3B588EE3AFB]:0) |  
  | REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT.testOutlierDetectionStopAndRestart" -Dtests.seed=591EC3B588EE3AFB -Dtests.security.manager=true -Dtests.locale=id-ID -Dtests.timezone=Europe/Moscow -Dcompiler.java=14 |
```

Reproduction line:

`/gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT.testOutlierDetectionStopAndRestart" -Dtests.seed=591EC3B588EE3AFB -Dtests.security.manager=true -Dtests.locale=id-ID -Dtests.timezone=Europe/Moscow -Dcompiler.java=14`

I couldn't reproduce it locally.

Sorry, need to reopen this at least for 7.x, there still similar looking failures although the branch should contain the fix PR.
e.g.  https://gradle-enterprise.elastic.co/s/c7hhi3j24h7o6

has:
```
org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT > testOutlierDetectionStopAndRestart FAILED
    java.lang.Exception: Test abandoned because suite timeout was reached.
        at __randomizedtesting.SeedInfo.seed([976F90FC40186C5D]:0)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT.testOutlierDetectionStopAndRestart" -Dtests.seed=976F90FC40186C5D -Dtests.security.manager=true -Dtests.locale=ar -Dtests.timezone=America/Guyana -Dcompiler.java=14

org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT > classMethod FAILED
    java.lang.Exception: Suite timeout exceeded (>= 1200000 msec).
        at __randomizedtesting.SeedInfo.seed([976F90FC40186C5D]:0)
```

and subsequent tests seem to fail as a result with active tasks like:
```
org.elasticsearch.xpack.ml.integration.MlJobIT > testMultiIndexDelete FAILED
    java.lang.AssertionError: 2 active tasks found:
```

There's at least another instance where it looks like a different test times out (ClassificationIT > testStopAndRestart). Unsure if thats related:
```
org.elasticsearch.xpack.ml.integration.ClassificationIT > testStopAndRestart FAILED
    java.lang.Exception: Test abandoned because suite timeout was reached.
        at __randomizedtesting.SeedInfo.seed([62320D5F6614A93E]:0)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.ClassificationIT.testStopAndRestart" -Dtests.seed=62320D5F6614A93E -Dtests.security.manager=true -Dtests.locale=fr-BE -Dtests.timezone=Asia/Rangoon -Dcompiler.java=14
```

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java
@@ -225,7 +225,7 @@ public class DataFrameAnalyticsManager {
                 startAnalytics(task, config);
             },
             error -> {
-                if (error instanceof TaskCancelledException && task.isStopping()) {
+                if (ExceptionsHelper.unwrapCause(error) instanceof TaskCancelledException && task.isStopping()) {
                     LOGGER.debug(new ParameterizedMessage("[{}] Caught task cancelled exception while task is stopping",
                         config.getId()), error);
                     task.markAsCompleted();

